src/openrct2/object/RideObject.cpp

```diff
@@ -76,26 +78,18 @@ void RideObject::ReadLegacy(IReadObjectContext * context, IStream * stream)
     GetStringTable()->Read(context, stream, OBJ_STRING_ID_NAME);
     GetStringTable()->Read(context, stream, OBJ_STRING_ID_DESCRIPTION);
 
-    // TODO: Move to its own function when ride construction window is merged.
-    if (gConfigInterface.select_by_track_type)
+    // Add boosters if the track type is eligible
+    for (sint32 i = 0; i < 3; i++)
     {
-        _legacyType.enabledTrackPieces = 0xFFFFFFFFFFFFFFFF;
-    }
-    else
-    {
-        // When not in select by track type mode, add boosters if the track type is eligible
-        for (sint32 i = 0; i < 3; i++)
-        {
-            if (_legacyType.ride_type[i] == RIDE_TYPE_LOOPING_ROLLER_COASTER ||
-                _legacyType.ride_type[i] == RIDE_TYPE_CORKSCREW_ROLLER_COASTER ||
-                _legacyType.ride_type[i] == RIDE_TYPE_TWISTER_ROLLER_COASTER ||
-                _legacyType.ride_type[i] == RIDE_TYPE_VERTICAL_DROP_ROLLER_COASTER ||
-                _legacyType.ride_type[i] == RIDE_TYPE_GIGA_COASTER ||
-                _legacyType.ride_type[i] == RIDE_TYPE_JUNIOR_ROLLER_COASTER)
-            {
-
-                _legacyType.enabledTrackPieces |= (1ULL << TRACK_BOOSTER);
-            }
+        if (
+            _legacyType.ride_type[i] == RIDE_TYPE_LOOPING_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_CORKSCREW_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_TWISTER_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_VERTICAL_DROP_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_GIGA_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_JUNIOR_ROLLER_COASTER
+        ) {
```

Comment:
It's unrelated, and I don't like snowballing this PR even further, even though I agree otherwise. I have created issue #5613 for it.